ios/RNGeth.swift

```diff
@@ -82,16 +109,201 @@ class RNGeth: RCTEventEmitter, GethNewHeadHandlerProtocol {
+     *
+     * @param data String hex encoded input
+     * @param signer String signer address
+     * @param passphrase String the account passphrase
+     * @param promise Promise
+     * @return Return signed transaction
+     */
+    @objc(signHashPassphrase:signer:passphrase:resolver:rejecter:)
+    func signHashPassphrase(data: NSString, signer: NSString, passphrase: NSString, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
+        do {
+            guard let keyStore = geth_node.getKeyStore() else {
+                throw RuntimeError("KeyStore not ready")
+            }
+            let data = Data(hexString: data as String)
+            let signer = try geth_node.findAccount(rawAddress: signer as String)!
+            let signature = try keyStore.signHashPassphrase(signer, passphrase: passphrase as String, hash: data)
+            resolve([signature.hex(prefixed:true)] as NSObject)
+        } catch let NSErr as NSError {
+            NSLog("@", NSErr)
+            reject(nil, nil, NSErr)
+        }
+    }
+
+    /**
+     * Signs a transaction using an unlocked account
+     *
+     * @param txRLP Data RLP encoded transaction
+     * @param from Stirng signer address
+     * @param promise Promise
+     * @return Return signed transaction
+     */
+    @objc(signTransaction:signer:resolver:rejecter:)
+    func signTransaction(txRLPHex: NSString, signer: NSString, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
+        var error: NSError?
+        do {
+            guard let keyStore = geth_node.getKeyStore() else {
+                throw RuntimeError("KeyStore not ready")
+            }
+            let txRLP = Data(hexString: txRLPHex as String)
+            guard let tx = GethNewTransactionFromRLP(txRLP, &error) else {
+                throw error ?? RuntimeError("Unable to create tx from RLP")
+            }
+            let signer = try geth_node.findAccount(rawAddress: signer as String)!
+            let chainID = GethNewBigInt(geth_node.getNodeConfig()?.ethereumNetworkID ?? 0)
+            let signedTx = try keyStore.signTx(signer, tx: tx, chainID: chainID)
+            let encodedTx = try signedTx.encodeRLP()
+            resolve([encodedTx.hex(prefixed:true)] as NSObject)
+        } catch let NSErr as NSError {
+            NSLog("@", NSErr)
+            reject(nil, nil, NSErr)
+        }
+    }
+
+    /**
+     * Signs a transaction using a passphrase.
+     *
+     * @param txRLP Data RLP encoded transaction
+     * @param from Stirng signer address
+     * @param passphrase String passphrase
+     * @param promise Promise
+     * @return Return signed transaction
+     */
+    @objc(signTransactionPassphrase:signer:passphrase:resolver:rejecter:)
+    func signTransactionPassphrase(txRLPHex: NSString, signer: NSString, passphrase: NSString, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
```

Comment:
I hesitate to move this logic into a native library which doesn't have much visibility or test coverage

Reply:
I agree with the lack of testing, but all this is actually doing is calling into the code inside of `celo-blockchain` in the `mobile/accounts.go` where the KeyStore that gets exposed to this native bridge is implemented. So essentially 99% of the logic here is just calling into the linked Go code.

That being said I do agree that some testing would be in order. I'll think about that.